src/clr-angular/popover/common/abstract-popover.ts

```diff
@@ -114,11 +114,13 @@ export abstract class AbstractPopover implements AfterViewChecked, OnDestroy {
   private ignore: any;
 
   private attachESCListener(): void {
-    this.documentESCListener = this.renderer.listen('document', 'keydown', event => {
-      if (event && event.keyCode === ESC) {
-        this.ifOpenService.open = false;
-      }
-    });
+    if (!this.popoverOptions.ignoreGlobalESCListener) {
+      this.documentESCListener = this.renderer.listen('document', 'keydown', event => {
+        if (event && event.keyCode === ESC) {
+          this.ifOpenService.open = false;
+        }
+      });
+    }
```

Comment:
Essentially, the current version of popover listens for ESC key on document and closes them. This needs to be overridden for dropdown menu as the ESC key should only close the current popover and move the cursor to its parent.